tests/containers/docker_image.pm

```diff
@@ -30,83 +30,22 @@ sub run {
     $self->select_serial_terminal;
 
     my ($image_names, $stable_names) = get_suse_container_urls();
-    my ($plugin, $osversion, $version);
 
+    ensure_ca_certificates_suse_installed() if (is_sle());
     install_docker_when_needed();
 
-    if (is_sle && !is_public_cloud) {
-        # Allow our internal 'insecure' registry
-        assert_script_run("mkdir -p /etc/docker");
-        assert_script_run('cat /etc/docker/daemon.json; true');
-        assert_script_run(
-            'echo "{ \"insecure-registries\" : [\"registry.suse.de\", \"registry.suse.de:443\", \"registry.suse.de:5000\"] }" > /etc/docker/daemon.json');
-        assert_script_run('cat /etc/docker/daemon.json');
-        systemctl('restart docker');
-
-        if (get_var('SCC_DOCKER_IMAGE')) {
-            my $regcode = get_var 'SCC_DOCKER_IMAGE';
-            assert_script_run "cp /etc/zypp/credentials.d/SCCcredentials{,.bak}";
-            assert_script_run "echo -ne \"$regcode\" > /etc/zypp/credentials.d/SCCcredentials";
-        }
-    }
-
-    if (check_var("ARCH", "x86_64") && !is_public_cloud) {
-        zypper_call("install container-diff");
-    }
+    allow_registry_suse_de()       if (is_sle && !is_public_cloud);
+    scc_docker_image_credentials() if (get_var('SCC_DOCKER_IMAGE'));
```

Comment:
Shouldn't it be `scc_docker_image_credentials() if (is_sle && !is_public_cloud && get_var('SCC_DOCKER_IMAGE'));` ?

Reply:
Why? I don't think so. Look at [container-suseconnect#building-images-on-non-sle-distributions](https://github.com/SUSE/container-suseconnect#building-images-on-non-sle-distributions)

Reply:
I was looking at the original `docker_image.pm` and there `scc_docker_image_credentials()` is executed in the `if (is_sle && !is_public_cloud) {` block.

https://github.com/SUSE/container-suseconnect#building-images-on-non-sle-distributions suggests that this is indeed not needed there.

Reply:
It is not needed. And the `SCC_DOCKER_IMAGE` itself is used rarely only in our scenarios where we need it anyways 😄